### DISCUSSION TITLE
[QPROF] Load sql file with QueryProfiler

### DISCUSSION THREAD

**Discussion by yakirgb** [vertica/VerticaPy#1292] [Q&A] [answered]
Hi,
Is it possible to execute SQL file direct from `QueryProfiler`?
I mean, instead create SQL file with setting of resource pool like:

```
head /Users/yakir.g/query.sql
SET SESSION RESOURCE_POOL = 'api_pool';
PROFILE WITH...SELECT
```
And  call it using; 
```
%load_ext verticapy.sql
%sql -nrows 1 -f /Users/yakir.g/query.sql
```

And take manual the values of `transaction_id` and `statement_id` from output into `QueryProfiler`

I'm looking for something like:

```
qprof = QueryProfiler(
sql_file = '/Users/yakir.g/query.sql'
resource_pool = 'api_pool'
..
)
```

Thank you, Yakir Gibraltar

**Comment by oualib** [comment]
Hi @yakirgb Nice idea, I'll see what we can do.

**Comment by oualib** [comment, marked as answer]
An issue has been created to create your params: https://github.com/vertica/VerticaPy/issues/1303

**Comment by oualib** [comment]
Done: https://github.com/vertica/VerticaPy/pull/1302

**Reply by yakirgb** [reply]
wow, very fast! 
Tested and it's working as expected. 
Thank you @oualib
